osf/management/commands/check_crossref_dois.py

```diff
@@ -66,31 +65,43 @@ def check_crossref_dois(dry_run=True):
 
         pending_dois = []
         for preprint in preprint_batch:
-            prefix = preprint.provider.doi_prefix
-            pending_dois.append(f'doi:{settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)}')
+            doi_prefix = preprint.provider.doi_prefix
+            if not doi_prefix:
+                sentry.log_message(f'Preprint [_id={preprint._id}] has been skipped for CrossRef DOI Check '
+                                   f'since the provider [_id={preprint.provider._id}] has invalid DOI Prefix '
+                                   f'[doi_prefix={doi_prefix}]')
+                continue
+            pending_dois.append(f'doi:{settings.DOI_FORMAT.format(prefix=doi_prefix, guid=preprint._id)}')
 
         url = '{}works?filter={}'.format(settings.CROSSREF_JSON_API_URL, ','.join(pending_dois))
 
         try:
             resp = requests.get(url)
             resp.raise_for_status()
         except requests.exceptions.HTTPError as exc:
-            logger.error(f'Could not contact crossref to check for DOIs, response returned with exception {exc}')
-            raise exc
+            sentry.log_message(f'Could not contact crossref to check for DOIs, response returned with exception {exc}')
+            continue
 
         preprints_response = resp.json()['message']['items']
 
         for preprint in preprints_response:
-            guid = preprint['DOI'].split('/')[-1]
-            base_guid, version = guid.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)
-            pending_preprint = preprints_with_pending_dois.get(
+            preprint__id = preprint['DOI'].split('/')[-1]
+            base_guid, version = Guid.split_guid(preprint__id)
+            if not base_guid or not version:
+                sentry.log_message(f'[Skipped] Preprint [_id={preprint__id}] returned by CrossRef API has invalid _id')
+                continue
```

Comment:
For reviewer: CrossRef must return the correct full (base and version) `_id`. Similarly, don't quit but just skip if one fails.